src/osdatahub/NGDAPI/ngd_api.py

```diff
@@ -0,0 +1,117 @@
+import json
+from datetime import datetime
+from os import environ
+from typing import Union
+
+import requests
+from geojson import FeatureCollection
+from typeguard import check_argument_types
+
+from osdatahub import Extent
+from osdatahub.NGDAPI.crs import get_crs
+from osdatahub.errors import raise_http_error
+
+
+def merge_geojsons(gj1, gj2):
+    if not (gj1 or gj2):
+        raise ValueError("Inputs were both empty")
+    elif not gj1:
+        return gj2
+    elif not gj2:
+        return gj1
+
+    gj1["features"] += gj2["features"]
+    gj1["numberReturned"] += gj2["numberReturned"]
+    gj1["links"] += gj2["links"]
+
+    return gj1
+
+
+class NGDAPI:
+    __ENDPOINT = r"https://api.os.uk/features/ngd/ofa/v1/"
+
+    DEFAULTS = {
+
+    }
+
+    def __init__(self, key: str, collection: str, extent: Extent = None):
+        self.key: str = key
+        self.collection: str = collection
+        self.extent = extent
+
+    # @property
+    # def collection(self):
+    #     return self.__collection
+    #
+    # @collection.setter
+    # def collection(self, col: str):
+    #     col = validate_collection(col)
+    #     self.__collection = col
+    #     self.__collection_name = col
+
+    def __endpoint(self, collection, feature_id=None) -> str:
+        return f"{self.__ENDPOINT}/collections/{collection}/items/{feature_id if feature_id else ''}"
+
+    def query(self,
+              extent: Extent = None,
```

Comment:
Would this construct a filter if the extent wasn't a bbox? Otherwise, would it be useful to call it bbox or bbox_extent or something?

Reply:
Yeah we weren't sure about this one either. One idea could be just to not use the bbox parameter and use the intersects CQL filter to make any extent filter valid